File: app/src/main/java/com/m7mdabaza/fooddelivery/ui/adapters/ProfileRecyclerAdapter.java
package com.m7mdabaza.fooddelivery.ui.adapters;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.m7mdabaza.fooddelivery.R;
import com.m7mdabaza.fooddelivery.ui.models.ProfileModel;

import java.util.ArrayList;

public class ProfileRecyclerAdapter extends RecyclerView.Adapter<ProfileRecyclerAdapter.ProfileViewHolder> {

    private ArrayList<ProfileModel> ProfileModelList = new ArrayList<>();


    @NonNull
    @Override
    public ProfileViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.profile_recycler_item, parent, false);
        return new ProfileViewHolder(view);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull ProfileViewHolder holder, int position) {

        ProfileModel profileModel = ProfileModelList.get(position);
        holder.profileInfo.setText(profileModel.getProfileInfo());
    }

    @Override
    public int getItemCount() {
        return ProfileModelList.size();
    }

    static class ProfileViewHolder extends RecyclerView.ViewHolder {

        TextView profileInfo;

        ProfileViewHolder(@NonNull View itemView) {
            super(itemView);
            profileInfo = itemView.findViewById(R.id.profileInfo_item);
        }

    }

    public void setList(ArrayList<ProfileModel> ProfileModel) {
        this.ProfileModelList = ProfileModel;
    }
}

File: app/src/main/java/com/m7mdabaza/fooddelivery/adapters/ProfileRecyclerAdapter.kt
package com.m7mdabaza.fooddelivery.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.m7mdabaza.fooddelivery.R
import com.m7mdabaza.fooddelivery.models.ProfileModel
import com.m7mdabaza.fooddelivery.ui.activities.HomeActivity
import java.util.*

class ProfileRecyclerAdapter : RecyclerView.Adapter<ProfileRecyclerAdapter.ProfileViewHolder>() {

    private var profileModelList = ArrayList<ProfileModel>()
    private lateinit var mContext: HomeActivity

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProfileViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.profile_recycler_item, parent, false)
        return ProfileViewHolder(view)
    }

    override fun onBindViewHolder(holder: ProfileViewHolder, position: Int) {
        val profileModel = profileModelList[position]
        holder.profileInfo.text = profileModel.profileInfo

    }

    override fun getItemCount(): Int {
        return profileModelList.size
    }

    class ProfileViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var profileInfo = itemView.findViewById(R.id.profileInfo_item) as TextView
    }

    fun setList(ProfileModel: ArrayList<ProfileModel>, mContext: HomeActivity) {
        this.profileModelList = ProfileModel
        this.mContext = mContext
    }
}